### PR TITLE
Start Android Components 90.0.0 cycle.

### PR DESCRIPTION
**Pull request by pocmo**
This bumps the version and updates the changelog to start the Android Components 90.0.0 cycle.

This PR also contains the changes required in our repository to implement RFC 7: https://mozac.org/rfc/0007-synchronized-releases (#10070). With that we will only have one `browser-engine-gecko` component following GeckoView Nightly (90) on `master`.

### CODE REVIEW

File: taskcluster/ci/build/kind.yml [hunk 1]
```diff
@@ -130,16 +130,6 @@ job-defaults:
...
-    browser-engine-gecko-nightly:
-        treeherder:
-            symbol: 'browser-engine-gecko-nigh(B)'
-    nightly-browser-engine-gecko-nightly:
-        treeherder:
-            symbol: 'browser-engine-gecko-nigh(BN)'
-    release-browser-engine-gecko-nightly:
-        treeherder:
-            symbol: 'browser-engine-gecko-nigh(BR)'
-
```
Comment (Amejia481) [inline taskcluster/ci/build/kind.yml:142]: @pocmo , in this file we don't need anything related to gecko, the previous changes were only related to gecko nightly?

Comment (pocmo) [reply taskcluster/ci/build/kind.yml:142]: In this file we only needed to remove `-beta` and `-nightly` entries.

Comment (Amejia481) [reply taskcluster/ci/build/kind.yml:142]: 👍🏽

**Review by Amejia481 — APPROVED** [review]
Tested with locally with sample browser and Fenix and it's working like a charm 🚀 !

**Review by csadilek — APPROVED** [review]
👍 🥇

**Pull request merged**